src/Connection.php

```diff
@@ -1045,39 +1041,44 @@ public function transactional(Closure $func): mixed
     /**
      * Sets if nested transactions should use savepoints.
      *
+     * @deprecated No replacement planned
+     *
      * @throws Exception
      */
     public function setNestTransactionsWithSavepoints(bool $nestTransactionsWithSavepoints): void
     {
         if (! $nestTransactionsWithSavepoints) {
-            Deprecation::trigger(
-                'doctrine/dbal',
-                'https://github.com/doctrine/dbal/pull/5383',
-                <<<'DEPRECATION'
-                Nesting transactions without enabling savepoints is deprecated.
-                Call %s::setNestTransactionsWithSavepoints(true) to enable savepoints.
-                DEPRECATION,
-                self::class
-            );
-        }
-
-        if ($this->transactionNestingLevel > 0) {
-            throw MayNotAlterNestedTransactionWithSavepointsInTransaction::new();
+            throw new LogicException(sprintf(
+                'Calling %s does not have any effect, that method only exists for backward compatibility reasons.',
```

Comment:
I believe this is a bit misleading. Calling this method with `true` in DBAL 4 is logically valid, so the LogicException is inappropriate. The dummy mode which could be previously enabled with `false` is no longer supported, so it's an `InvalidArgumentException`, and the exception message should explicitly mention `false` and/or the dummy mode.